### PR TITLE
Fix disk_index with current disk more than 7 due to scsi controller

### PR DESCRIPTION
**Pull request by jmvandry**
##### SUMMARY

When you already have 7 disk added to your VM, you can't add another disk since the index is incremented. The validation is wrong because the disk_index and the number of disk is not ajusted correctly.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

vmware_guest
##### ADDITIONAL INFORMATION

Playbook:

 - name: "test new disk when 7 disk exist"
   hosts: all
   gather_facts: no
   tasks:
    - vmware_guest_disk_facts:
       hostname: "{{vmware_vcenter}}"
       username: "{{vmware_username}}"
       password: "{{vmware_password}}"
       validate_certs: no
       datacenter: "{{vmware_datacenter}}"
       name: "{{ inventory_hostname }}"
      delegate_to: localhost
      become: no
      register: vmware_add_disk_disk_facts

    - name: Reset existing_disk
      set_fact:
        vmware_add_disk_existing_disk: ""

    - name: Construire la liste des disques existants
      set_fact:
        vmware_add_disk_existing_disk: "{% if vmware_add_disk_existing_disk != '' %}{{vmware_add_disk_existing_disk}}+{% endif %}{{[{'size_kb': vmware_add_disk_disk_facts.guest_disk_facts[disk|string].capacity_in_kb,'type':'eagerzeroedthick'}]}}"
      when: vmware_add_disk_disk_facts.guest_disk_facts[disk|string] is defined
      loop: "{{range(0,15)|list}}"
      loop_control:
       loop_var: disk

    - vmware_guest:
       hostname: "{{vmware_vcenter}}"
       username: "{{vmware_username}}"
       password: "{{vmware_password}}"
       validate_certs: no
       datacenter: "{{vmware_datacenter}}"
       name: "{{ inventory_hostname }}"
       disk: "{{vmware_add_disk_existing_disk}} + {{[{'size_gb': 10,'type':'eagerzeroedthick'}]}}"
      delegate_to: localhost
      become: no

```paste below
Before change:
#ansible-playbook -i inventaire/nouvelle_vm.yml vmware-add-disk.yml

PLAY [test new disk when 7 disk exist] *******************************************************************************************************************************************************************************************************

TASK [vmware_guest_disk_facts] ***************************************************************************************************************************************************************************************************************
ok: [vm1 -> localhost]

TASK [Reset existing_disk] *******************************************************************************************************************************************************************************************************************
ok: [vm1]

TASK [Construire la liste des disques existants] *********************************************************************************************************************************************************************************************
ok: [vm1] => (item=0)
ok: [vm1] => (item=1)
ok: [vm1] => (item=2)
ok: [vm1] => (item=3)
ok: [vm1] => (item=4)
ok: [vm1] => (item=5)
ok: [vm1] => (item=6)
skipping: [vm1] => (item=7)
skipping: [vm1] => (item=8)
skipping: [vm1] => (item=9)
skipping: [vm1] => (item=10)
skipping: [vm1] => (item=11)
skipping: [vm1] => (item=12)
skipping: [vm1] => (item=13)
skipping: [vm1] => (item=14)

TASK [vmware_guest] **************************************************************************************************************************************************************************************************************************
changed: [vm1 -> localhost]

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
vm1 : ok=4    changed=1    unreachable=0    failed=0

# ansible-playbook -i inventaire/nouvelle_vm.yml vmware-add-disk.yml


PLAY [test new disk when 7 disk exist] *******************************************************************************************************************************************************************************************************

TASK [vmware_guest_disk_facts] ***************************************************************************************************************************************************************************************************************
ok: [vm1 -> localhost]

TASK [Reset existing_disk] *******************************************************************************************************************************************************************************************************************
ok: [vm1]

TASK [Construire la liste des disques existants] *********************************************************************************************************************************************************************************************
ok: [vm1] => (item=0)
ok: [vm1] => (item=1)
ok: [vm1] => (item=2)
ok: [vm1] => (item=3)
ok: [vm1] => (item=4)
ok: [vm1] => (item=5)
ok: [vm1] => (item=6)
ok: [vm1] => (item=7)
skipping: [vm1] => (item=8)
skipping: [vm1] => (item=9)
skipping: [vm1] => (item=10)
skipping: [vm1] => (item=11)
skipping: [vm1] => (item=12)
skipping: [vm1] => (item=13)
skipping: [vm1] => (item=14)

TASK [vmware_guest] **************************************************************************************************************************************************************************************************************************
fatal: [vm1 -> localhost]: FAILED! => {"changed": true, "msg": "Invalid configuration for device '0'."}
        to retry, use: --limit @/home/path/ansible/vmware-add-disk.retry

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
vm1 : ok=3    changed=0    unreachable=0    failed=1





After change:
# ansible-playbook -i inventaire/nouvelle_vm.yml  vmware-add-disk.yml

PLAY [test new disk when 7 disk exist] *******************************************************************************************************************************************************************************************************

TASK [vmware_guest_disk_facts] ***************************************************************************************************************************************************************************************************************
ok: [vm1 -> localhost]

TASK [Reset existing_disk] *******************************************************************************************************************************************************************************************************************
ok: [vm1]

TASK [Construire la liste des disques existants] *********************************************************************************************************************************************************************************************
ok: [vm1] => (item=0)
ok: [vm1] => (item=1)
ok: [vm1] => (item=2)
ok: [vm1] => (item=3)
ok: [vm1] => (item=4)
ok: [vm1] => (item=5)
ok: [vm1] => (item=6)
ok: [vm1] => (item=7)
skipping: [vm1] => (item=8)
skipping: [vm1] => (item=9)
skipping: [vm1] => (item=10)
skipping: [vm1] => (item=11)
skipping: [vm1] => (item=12)
skipping: [vm1] => (item=13)
skipping: [vm1] => (item=14)

TASK [vmware_guest] **************************************************************************************************************************************************************************************************************************
changed: [vm1 -> localhost]

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
vm1 : ok=4    changed=1    unreachable=0    failed=0

```

### CODE REVIEW

**Comment by Akasurde** [comment]
@jmvandry Thanks for the PR. Could you please add an [integration test](https://github.com/ansible/ansible/tree/devel/test/integration/targets/vmware_guest/tasks) for this change ? Thanks.

**Comment by jmvandry** [comment]
@Akasurde  I've add the integration test but it failed. I think it fail because the fix of disk_index as I proposed is not applied to branch devel.

File: test/integration/targets/vmware_guest/tasks/disk_max_ctrl_d1_c1_f0.yml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+- name: get a list of VMS from vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/govc_find?filter=VM
+  register: vmlist
+
+- debug: var=vcsim_instance
+- debug: var=vmlist
+
+- name: create new VMs with 14 disk
```
Comment (Akasurde) [inline test/integration/targets/vmware_guest/tasks/disk_max_ctrl_d1_c1_f0.yml:34]: Can you run this task again to check the idempotency of the module before deleting the VM created ?

**Comment by Akasurde** [comment]
@jmvandry There is error around 
```
02:48 TASK [vmware_guest : assert that changes were made] ****************************
02:48 fatal: [testhost]: FAILED! => {
02:48     "assertion": "disk_max_ctrl_d1_c1_f0_add_disk_8.results|map(attribute='changed')|unique|list == [true]",
02:48     "changed": false,
02:48    "evaluated_to": false,
02:48     "msg": "Assertion failed"
02:48
02:48 	to retry, use: --limit @/root/ansible/test/integration/vmware_guest-xwebie83.retry
```

**Comment by jmvandry** [comment]
@Akasurde I can't download the output of shippable to understand why there was nos changed, if you look at the integration test, it should have changed. Has you can view it add a disk but the changed state is false, which I can't understand, could you look why or give me the ouput ?

**Comment by jmvandry** [comment]
This is still a problem btw, i think it should be merge

**Review by pdellaert — APPROVED** [review]
LGTM

Good improvement :)

**Pull request closed without merging**